harmony/harmony.py

```diff
@@ -968,6 +985,124 @@ def download_all(self,
         for url in self.result_urls(job_id, show_progress=False) or []:
             yield self.executor.submit(self._download_file, url, directory, overwrite)
 
+    def iterator(
+        self,
+        job_id: str,
+        directory: str = '',
+        overwrite: bool = False
+    ) -> Iterator:
+        """Create an iterator that will poll for data in the background and download it as
+        it is available and requested via `next()`.
+
+        Each iteration returns a dictionary, or `None` when all granules have been iterated.
+        The dictionary has the following form:
+
+        {
+            'path': Future
+            'bbox': BBox object containing the bounding box for the granule,
+            'temporal': {
+                'start': '2020-01-11T14:00:00.000Z',
+                'end': '2020-01-11T15:59:59.000Z'
+            }
+        }
+
+        The Future resolves to the path to the downloaded file.
+
+        If the job is paused and all processed granules have already been returned in
+        the iteration, the status returned will be GranuleStatus.PAUSED until the job
+        is resumed.
+
+        If the job fails during iteration then calls to `next` will raise an exception.
+        Note that this is not true if the job completed with errors, in which case specific
+        granules may return errors, but no exception is raised. This allows the caller
+        to retrieve any granules that were successfully processed.
+
+        Note: if a job gets stuck in the 'running' state, this iterator will happily wait
+        forever re-checking the status page periodically.
+
+        Args:
+            job_id: UUID string for the job you wish to interrogate.
+            directory: Optional. If specified, saves files there. Saves files to the current
+            working directory by default.
+            overwrite: If True, will overwrite a local file that shares a filename with the
+            downloaded file. Defaults to False. If you're seeing malformed data or truncated
+            files from incomplete downloads, set overwrite to True.
+
+        Returns:
+            An Iterator that can be used to iterate over the granule results from a job
+        """
+        GET_JSON_RETRY_LIMIT = int(os.getenv('GET_JSON_RETRY_LIMIT', 3))
+        GET_JSON_RETRY_SLEEP = float(os.getenv('GET_JSON_RETRY_SLEEP', 1.0))
+        next_url = self._status_url(job_id)
+
+        # index used to keep track of where we are in the data links if the page gets reloaded
+        # so we don't pull the same granule more than once
+        # NOTE: this assumes that granule order is preserved when the page is reloaded and
```

Comment:
When what page is reloaded? Do you mean that it is relying on the order of the links returned by harmony job status pages is consistent? I think that assumption is fine, but if that's what we're relying on maybe change the comment to:
```
 # index used to keep track of where we are in the data links if the page gets reloaded
 # so we don't pull the same data links more than once
 # NOTE: this relies on the link order being preserved when the harmony job status page
 # is reloaded and new links are added.
```

Reply:
Done